sdk/src/main/scala/io/horizen/account/state/ForgerStakeStorage.scala

```diff
@@ -0,0 +1,358 @@
+package io.horizen.account.state
+
+import com.google.common.primitives.Bytes
+import io.horizen.account.proposition.AddressProposition
+import io.horizen.account.state.ForgerStakeLinkedList.{addNewNode, getStakeListItem, linkedListNodeRefIsNull, removeNode}
+import io.horizen.account.state.ForgerStakeStorage.saveStorageVersion
+import io.horizen.account.state.ForgerStakeStorageVersion.ForgerStakeStorageVersion
+import io.horizen.account.state.NativeSmartContractMsgProcessor.NULL_HEX_STRING_32
+import io.horizen.account.state.WithdrawalMsgProcessor.calculateKey
+import io.horizen.account.utils.BigIntegerUtil
+import io.horizen.account.utils.WellKnownAddresses.FORGER_STAKE_SMART_CONTRACT_ADDRESS
+import io.horizen.evm.Address
+import io.horizen.proposition.{PublicKey25519Proposition, VrfPublicKey}
+import sparkz.crypto.hash.Blake2b256
+
+import java.math.BigInteger
+import scala.util.{Failure, Success}
+
+trait ForgerStakeStorage {
+
+  def existsStakeData(view: BaseAccountStateView, stakeId: Array[Byte]): Boolean = {
```

Comment:
The same as below, maybe we can move it in the companion object

Reply:
See below